src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -745,18 +797,42 @@ await ExecuteWithFailOverPolicyAsync(
 
                 Dictionary<string, ConfigurationSetting> mappedData = await MapConfigurationSettings(data).ConfigureAwait(false);
                 SetData(await PrepareData(mappedData, cancellationToken).ConfigureAwait(false));
+
+                List<KeyValueWatcher> selectedKeyValueWatchers = new List<KeyValueWatcher>();
+
+                if (_options.RefreshOptions.RegisterAllEnabled)
+                {
+                    foreach (KeyValueIdentifier keyValueIdentifier in watchedSelectedKeyValueCollections.Keys)
+                    {
+                        selectedKeyValueWatchers.AppendUnique(new KeyValueWatcher()
+                        {
+                            Key = keyValueIdentifier.Key,
+                            Label = keyValueIdentifier.Label,
+                            RefreshInterval = _options.RefreshOptions.RefreshInterval
+                        });
+                    }
+                }
+
+                // Update the next refresh time for all refresh registered settings and feature flags
+                foreach (KeyValueWatcher changeWatcher in _options.ChangeWatchers.Concat(_options.FeatureFlagWatchers).Concat(selectedKeyValueWatchers))
+                {
+                    UpdateNextRefreshTime(changeWatcher);
+                }
+
                 _watchedSettings = watchedSettings;
+                _watchedSelectedKeyValueCollections = watchedSelectedKeyValueCollections;
+                _watchedFeatureFlagCollections = watchedFeatureFlagCollections;
                 _mappedData = mappedData;
+                _selectedKeyValueWatchers = selectedKeyValueWatchers;
             }
         }
 
-        private async Task<Dictionary<string, ConfigurationSetting>> LoadSelectedKeyValues(ConfigurationClient client, CancellationToken cancellationToken)
+        private async Task<Dictionary<KeyValueIdentifier, IEnumerable<MatchConditions>>> LoadSelected(Dictionary<string, ConfigurationSetting> existingData, IEnumerable<KeyValueSelector> selectors, ConfigurationClient client, CancellationToken cancellationToken)
         {
-            var serverData = new Dictionary<string, ConfigurationSetting>(StringComparer.OrdinalIgnoreCase);
+            Dictionary<KeyValueIdentifier, IEnumerable<MatchConditions>> watchedCollections = new Dictionary<KeyValueIdentifier, IEnumerable<MatchConditions>>();
```

Comment:
We could use `var` here to keep the line shorter and more readable. The type is clear from initialization.